language/move-lang/src/parser/ast.rs

```diff
@@ -529,7 +608,17 @@ impl BinOp_ {
             | B::Lt
             | B::Gt
             | B::Le
-            | B::Ge => true,
+            | B::Ge
+            | B::Range
+            | B::Implies => true,
+        }
+    }
+
+    pub fn is_spec_only(&self) -> bool {
```

Comment:
This function is unused now, right? If so, please remove it.

Reply:
Argh, missed this one. The function is still used, but now in expansion, where we are checking whether an operator is allowed in programs.

Reply:
Maybe the function can move to expansion/ast.rs then?

Reply:
The type BinOp is defined in the parser/ast.rs, and already has similar functions (is_pure IIRC). Some types from the parser AST are reused in later phases, specifically those basic enum types. So I think it is inline with the existing code style to leave it there.